### PR TITLE
Iterate keys of object when used with `for...in` loop

### PR DESCRIPTION
**Pull request by aleclarson**
This allows `for...in` loops to work with `{}` or `Object.create(null)`.

I added `isObject` and `isEmpty` for cleaner `if` statements. I can remove those if you prefer.

### CODE REVIEW

File: tags/for.js [hunk 1]
```diff
@@ -38,8 +38,10 @@ module.exports = function (liquid) {
     render: function (scope, hash) {
       var collection = Liquid.evalExp(this.collection, scope)
 
-      if (!Array.isArray(collection) ||
-                (Array.isArray(collection) && collection.length === 0)) {
+      if (isObject(collection)) {
```
Comment (harttle) [inline tags/for.js]: Can you move these language related functions to [/src/util/underscore.js][underscore-local].

I don't mind renaming `underscore.js` to `lang.js` or something more appropriate if you find this file name confusing, cause it's actually not the [underscore][underscore] libarary.

[underscore-local]: https://github.com/harttle/shopify-liquid/blob/master/src/util/underscore.js
[underscore]: http://underscorejs.org/

File: tags/for.js [hunk 2]
```diff
@@ -90,3 +92,18 @@ module.exports = function (liquid) {
...
+
+function isObject(collection) {
+  if (collection) {
+    const ctr = collection.constructor
+    return ctr === Object || ctr == null
+  }
+  return false
+}
+
+function isEmpty(collection) {
```
Comment (harttle) [inline tags/for.js]: `isEmpty` should also capable of none-empty objects.

The underscore/lodash can be implementation references, e.g. <https://lodash.com/docs/4.17.4#isEmpty>

**Comment by aleclarson** [comment]
I added `isPlainObject` to `src/util/underscore.js` and removed `isEmpty` in favor of just simplifying the `if` statement.

`{}` and `Object.create(null)` with no keys are accounted for, since they are converted to arrays.

**Comment by harttle** [comment]
I've merged your commits and added test cases. Behaviors related to empty and non-empty strings will be compliant with [the ruby version](https://github.com/Shopify/liquid).

This feature is released as 1.9.0. Thank you to lead this feature.

**Pull request merged**